platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -977,11 +977,11 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
     }
 
     // bootclasspath has been removed in Java 9+
-    @Requires(UnitTestPreconditions.Jdk8OrEarlier)
+    @Requires(IntegTestPreconditions.BestJreAvailable)
```

Comment:
🤔 How do we make sure this still requires a Java 8 installation to execute?

Reply:
We didn't really care if we were running against a JDK 8. We cared if we were running against a JDK with a `$home/jre/lib/rt.jar`. `BestJreAvailable` checks to see if we have a JVM with a jre, which will have an `rt.jar`.